Use our own scheduling when doing d2d backups (experimental)

grote opened the pull request:
Biggest part of #583

grote committed:
Use our own scheduling when doing d2d backups (experimental)

grote committed:
Show success notification in a separate notification channel

so the user can disable it. This will be more of an issue now that it comes also for scheduled background backups.

grote commented:
We still need to check AOSP build.

grote committed:
Remove logWorkInfo as some part of it isn't in AOSP version of workmanger

It was meant for debugging only anyway

chirayudesai commented:
> We still need to check AOSP build.

Builds, untested

https://laskija.eu.calyxos.net/seedvault/610/e5be542abff2bca872220d5a738ef8a529ddbd3d.apk

grote committed:
Remove warning from d2d expert setting as we now disable system scheduling for d2d

grote committed:
Fix flaky metadata decoding test

grote committed:
Use own notification ID for success notification

and cancel observer notification when backup is finished. This is because when the success channel is disabled by the user, the observer notification could stick around and would never get removed.

stevesoltys commented on app/src/main/java/com/stevesoltys/seedvault/BackupWorker.kt:
```diff
@@ -0,0 +1,57 @@
+                .setRequiredNetworkType(NetworkType.UNMETERED)
+                .setRequiresCharging(true)
+                .build()
+            val backupWorkRequest = PeriodicWorkRequestBuilder<BackupWorker>(
```
This should be configurable when this becomes non-experimental.
